waku/v2/peermanager/service_slot.go

```diff
@@ -0,0 +1,75 @@
+package peermanager
+
+import (
+	"sync"
+
+	"github.com/libp2p/go-libp2p/core/peer"
+	"github.com/libp2p/go-libp2p/core/protocol"
+	"github.com/waku-org/go-waku/waku/v2/utils"
+)
+
+type peerMap struct {
+	mu sync.RWMutex
+	m  map[peer.ID]struct{}
+}
+
+func newPeerMap() *peerMap {
+	return &peerMap{
+		m: map[peer.ID]struct{}{},
+	}
+}
+
+func (pm *peerMap) getRandom() (peer.ID, error) {
```

Comment:
This is not using randomselection logic. 
It is just returning always first one. Need to fix this to return a random one from list of peers in serviceSlots.

Reply:
Interesting, was not aware of this part. 
Interesting they enforce randomness in iterator.